exporters/otlp/otlphttp/options.go

```diff
@@ -0,0 +1,141 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otlphttp
+
+import (
+	"crypto/tls"
+	"time"
+)
+
+// Compression describes the compression used for payloads sent to the
+// collector.
+type Compression int
+
+const (
+	// NoCompression tells the driver to send payloads without
+	// compression.
+	NoCompression Compression = iota
+	// GzipCompression tells the driver to send payloads after
+	// compressing them with gzip.
+	GzipCompression
+)
+
+const (
+	// DefaultMaxAttempts describes how many times the driver
+	// should retry the sending of the payload in case of a
+	// retryable error.
+	DefaultMaxAttempts int = 5
+	// DefaultTracesPath is a default URL path for endpoint that
+	// receives spans.
+	DefaultTracesPath string = "/v1/traces"
+	// DefaultMetricsPath is a default URL path for endpoint that
+	// receives metrics.
+	DefaultMetricsPath string = "/v1/metrics"
+	// DefaultBackoff is a default base backoff time used in the
+	// exponential backoff strategy.
+	DefaultBackoff time.Duration = 300 * time.Millisecond
+)
+
+type config struct {
+	address        string
+	compression    Compression
+	tracesURLPath  string
+	metricsURLPath string
+	maxAttempts    int
+	backoff        time.Duration
+	tlsCfg         *tls.Config
+	insecure       bool
+	headers        map[string]string
+}
+
+// Option applies an option to the HTTP driver.
+type Option func(cfg *config)
```

Comment:
Should we take this opportunity to convert `Option` into an interface to bring it in line with our [style guide](https://github.com/open-telemetry/opentelemetry-go/blob/master/CONTRIBUTING.md#option)?

Reply:
Yeah, will do.

Reply:
Done.